tangelo/algorithms/variational/iqcc_ilc_solver.py

```diff
@@ -106,6 +109,25 @@ def __init__(self, opt_dict):
             else:
                 raise KeyError(f"The keyword {param} is not available in self.__class__.__name__.")
 
+        copt_dict = opt_dict.copy()
+
+        self.molecule: SecondQuantizedMolecule = copt_dict.pop("molecule", None)
+        self.qubit_mapping: str = copt_dict.pop("qubit_mapping", "jw")
+        self.up_then_down: str = copt_dict.pop("up_then_down", False)
+        self.initial_var_params: Union[List, str] = copt_dict.pop("initial_var_params", None)
+        self.backend_options: dict = copt_dict.pop("backend_options", default_backend_options)
+        self.penalty_terms: Optional[dict] = copt_dict.pop("penalty_terms", None)
+        self.ilc_ansatz_options: dict = copt_dict.pop("ilc_ansatz_options", dict())
+        self.qcc_ansatz_options: dict = copt_dict.pop("qcc_ansatz_options", dict())
+        self.qubit_hamiltonain: QubitOperator = copt_dict.pop("qubit_hamiltonian", None)
```

Comment:
self.qubit_hamilton**ian** (you have a typo).

I'm a bit scared here because the rest of the code relies on `self.qubit_hamiltonian` and no error in our tests was caught.

Reply:
The error was caught in testing when I removed the old assignment from dictionary method.